crates/uv-resolver/src/lock/mod.rs

```diff
@@ -2837,6 +2837,34 @@ struct PackageMetadata {
     dependency_groups: BTreeMap<GroupName, BTreeSet<Requirement>>,
 }
 
+impl PackageMetadata {
+    fn unwire(self, requires_python: &RequiresPython) -> PackageMetadata {
+        // We need to complexify these markers so things like
+        // `requires_python < '0'` get normalized to False
+        let unwire_requirements = |requirements: BTreeSet<Requirement>| -> BTreeSet<Requirement> {
+            requirements
+                .into_iter()
+                .map(|mut requirement| {
+                    let complexified_marker =
+                        requires_python.complexify_markers(requirement.marker);
+                    requirement.marker = complexified_marker;
+                    requirement
+                })
+                .collect()
+        };
+
+        PackageMetadata {
+            requires_dist: self.requires_dist,
```

Comment:
Don't we need to `unwire` the `requires_dist` too?

Reply:
I have that in an earlier commit and it caused even more test failures. I imagine yes it wants the same treatment as these dependency-groups, whatever the correct treatment is.

Reply:
I could confirm this with the case below (failing on the branch), we should solve that one too.

```toml
[project]
name = "debug"
version = "0.1.0"
requires-python = ">=3.11"
dependencies = [
  "pytest; python_full_version>'3.8' and python_full_version<'3.6'"
]
```